### PR TITLE
CLNT_CALL with clnt_req

### PR DESCRIPTION
**Pull request by was4**
CLNT_CALL with clnt_req
Remove unusable clnt_control timeouts

### CODE REVIEW

**Comment by mattbenjamin** [comment]
looks fine

**Comment by was4** [comment]
Added inline clnt_req_fill() so the diffs don't look so large, but the generated code should be fast.

Changed the timeouts to allow 0 and don't wait on no timeout per Frank's request.

File: ntirpc/rpc/clnt.h [hunk 1]
```diff
@@ -515,10 +506,27 @@ clnt_dg_ncreate(const int fd, const struct netbuf *raddr,
...
 extern CLIENT *clnt_raw_ncreate(rpcprog_t, rpcvers_t);
 
+/*
+ * Client request processing
+ */
 int clnt_req_xid_cmpf(const struct opr_rbtree_node *lhs,
 		      const struct opr_rbtree_node *rhs);
 
-struct clnt_req *clnt_req_alloc(CLIENT *, struct timeval);
+static inline void clnt_req_fill(struct clnt_req *cc, struct rpc_client *clnt,
```
Comment (dang) [inline ntirpc/rpc/clnt.h:515]: So, it seems that clnt_req_fill() and clnt_req_setup() are always called as a pair.  Is there a reason to have separate calls for this?

**Comment by was4** [comment]
On 11/2/17 10:28 AM, Daniel Gryniewicz wrote:
> *@dang* commented on this pull request.
> 
> ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
> 
> In ntirpc/rpc/clnt.h <https://github.com/nfs-ganesha/ntirpc/pull/89#discussion_r148544049>:
> 
>>  int clnt_req_xid_cmpf(const struct opr_rbtree_node *lhs,
>   		      const struct opr_rbtree_node *rhs);
>   
> -struct clnt_req *clnt_req_alloc(CLIENT *, struct timeval);
> +static inline void clnt_req_fill(struct clnt_req *cc, struct rpc_client *clnt,
> 
> So, it seems that clnt_req_fill() and clnt_req_setup() are always called as a pair. Is there a reason to have separate calls for this?
> 
The former was added later for convenience and is inline.  It is called
only once per struct.

The latter is not inline -- the function is too large.  And it may be
called more than once.  Also allows variables to be set between them
(as is done in several places).

The whole purpose of this patch is to separate the underlying functions
without changing logic (for easier review).  Logic changes will be later.

For example, you'll note that CLNT_CALL functions now have only the
transport-specific calling code.  Next, the logic will be changed to
call the function again from the retry code.

**Review by dang — APPROVED** [review]

**Pull request merged**